.github/workflows/engine-nightly.yml

```diff
@@ -19,7 +19,7 @@ jobs:
     steps: [
       { name: 'Checkout', uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 },
       { name: 'Install Python', uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c, with: { python-version: 3.12, architecture: x64 } },
-      { name: 'XCode', uses: maxim-lobanov/setup-xcode@60606e260d2fc5762a71e64e74b2174e8ea3c8bd, with: { xcode-version: '15.1' } },
+      { name: 'XCode', uses: maxim-lobanov/setup-xcode@60606e260d2fc5762a71e64e74b2174e8ea3c8bd, with: { xcode-version: '15.4' } },
```

Comment:
Runner os version also should be updated because only macos-14 has Xcode 15.4 (https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md#xcode)

Reply:
thx!

Reply:
Interesting if it's only arm64 runners now

Reply:
Hmm, no. Arm and non-arm runner exists.
<img width="854" alt="Screenshot 2024-06-08 at 12 04 12" src="https://github.com/defold/defold/assets/770105/c4214e85-2711-48a1-9771-3b1100379d4a">

Reply:
Could you share a link to the page where did you make this screenshot?

Reply:
https://github.com/actions/runner-images/?tab=readme-ov-file#available-images

Reply:
thank you!